remark-prismjs: remove extra newline

docwhat opened the pull request:
Fixes #4802

docwhat commented:
DO NOT ACCEPT.  It isn't correct, yet.

KyleAMathews commented on packages/gatsby-remark-prismjs/src/highlight-code.js:
```diff
@@ -36,10 +36,12 @@ module.exports = (language, code, lineNumbersHighlight = []) => {
+    const maxCodeSplits = codeSplits.length - 1
+    codeSplits.forEach((split, idx) => {
+      highlightedCode += split.code
+      if (!split.highlighted && idx < maxCodeSplits) {
```
I don't understand this change — if I'm reading this right, you changed it so that only the last split that's not highlighted has a new line character added?

KyleAMathews replied:
Isn't the issue just we're adding a code split to InlineCode blocks and we never should do that? Is there some way of detecting if we're highlighting an InlineCode block?

docwhat commented:
The previous code added newly new lines to all non-highlighted code. It was an attempt to reverse the split on new lines.

The problem is it added an extra newline at the end. 

This fixes it.

KyleAMathews commented:
What was wrong with the previous pattern? The bug in #4802 only emerged with the inlinecode highlighting changes. This PR just stops adding back new lines altogether — which could be an acceptable fix.

docwhat commented:
> What was wrong with the previous pattern?

> The bug in #4802 only emerged with the inlinecode highlighting changes.

Correct.  That's because the previous code didn't pass `<code>` through `highlightCode()`.

> This PR just stops adding back new lines altogether — which could be an acceptable fix.

It doesn't stop all newlines.  It prevents an extra newline being added to the end.

The previous code did (roughly) this for the no-highlighted-lines case:

``` javascript
const codeSplits = highlightedCode.split(`\n`)

highlightedCode = ``
codeSplits.forEach(split => {
  highlightedCode += `${split.code}\n`
})
```

The problem is that this code above adds an additional trailing `\n`.

That trailing `\n` was the problem.  It was actually a problem _before_ my changes.  I had noticed the extra newline and added this to my list of things to fix.  I just got to it earlier than I thought. :smiley:

docwhat commented:
FYI: The appveyor check is barfing on an image-sharp thing. I don't think I did that.

KyleAMathews commented:
Fixed it in https://github.com/gatsbyjs/gatsby/pull/4823
